src/evaluators/utility/PHAL_IsAvailable.hpp

```diff
@@ -0,0 +1,46 @@
+//*****************************************************************//
+//    Albany 3.0:  Copyright 2016 Sandia Corporation               //
+//    This Software is released under the BSD license detailed     //
+//    in the file "license.txt" in the top-level Albany directory  //
+//*****************************************************************//
+
+#ifndef PHAL_IS_PARAM_AVAILABLE_HPP
+#define PHAL_IS_PARAM_AVAILABLE_HPP
+
+namespace PHAL
+{
+
+enum class FieldScalarType : int {
+  Real        = 0,
+  MeshScalar  = 1,
+  ParamScalar = 2,
+  Scalar      = 3
+};
+
+template<typename EvalT>
+Teuchos::RCP<PHX::FieldTag>
+createTag(const std::string& name, const Teuchos::RCP<PHX::DataLayout>& dl)
+{
+  return Teuchos::rcp(new PHX::Tag<typename EvalT::ScalarT>(name,dl));
+}
+
+template<typename EvalT>
+bool is_param_available (const PHX::FieldManager<PHAL::AlbanyTraits>& fm,
```

Comment:
It looks like this fcn can look for any kind of field, not just parameters. Wouldn't it be more apt to call it `is_field_evaluated`? Same change for the ifdef.